### PR TITLE
New package: maloc

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/maloc/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    memory management routines, timing routines, machine
+    epsilon, access to UNIX and INET sockets, MPI, and so on.
+    All things that can vary from one architecture to another
+    are abstracted out of an application code and placed in
+    MALOC. To port the application code to a new architecture,
+    only the small MALOC library needs to be ported (usually
+    just "./configure ; make"). MALOC takes the pain of varying
+    UNIX (and Win32) platforms with differing library and
+    header layous completely out of the software development
+    picture."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/maloc/package.py]: You probably don't need this entire description, the first sentence would do.

File: var/spack/repos/builtin/packages/maloc/package.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on('graphviz', type='build', when='+doc')
+    depends_on('doxygen', type='build', when='+doc')
+
+    def configure_args(self):
+        spec = self.spec
+        args = []
+
+        if '~doc' in spec:
+            args.append('--with-doxygen=no')
+            args.append('--with-dot=no')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/maloc/package.py:37]: Can you explicitly enable these when `+doc`?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/maloc/package.py (hunk 2 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/maloc/package.py:37]: So, those two flags don't *actually* do anything, even though `./configure --help` lists them as options. Should I write this as if they did something? Or should I remove the flags entirely? The variant exists more for the sake of removing python (by way of doxygen) from the list of dependencies (to fix problems with dependents wanting a mutually exclusive version of python. Listing as `type='build'` only doesn't solve this issue (at least in spack v0.13.2, what we're unfortunately stuck on at the moment)). Does it make sense to remove doxygen and dot from the list of dependencies and assume the user has them installed at the system level?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/maloc/package.py:37]: > Does it make sense to remove doxygen and dot from the list of dependencies and assume the user has them installed at the system level?

No, anything but this. If doxygen and dot are optional, then what you have now is fine. If they are required, they would always be dependencies. We should never assume anything is installed on the system.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**